samples/SoundBoard/src/main/java/com/google/oboe/samples/soundboard/MusicTileView.java

```diff
@@ -0,0 +1,143 @@
+package com.google.oboe.samples.soundboard;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.PointF;
+import android.graphics.Rect;
+import android.util.SparseArray;
+import android.view.MotionEvent;
+import android.view.View;
+
+import java.util.ArrayList;
+
+import static android.os.Process.getExclusiveCores;
+
+public class MusicTileView extends View {
+    private ArrayList<Rect> mRectangles;
+    private boolean[] mIsPressedPerRectangle;
+    private Paint mPaint;
+    private SparseArray<PointF> mLocationsOfFingers;
+
+    private long mEngineHandle = 0;
+    private native void setNoteOn(long engineHandle, int noteIndex);
```

Comment:
Rather than calling a native method directly, call a TileListener interface that has onTileOn(index) and onTimeOff(index) methods. Then you could reuse this for other use cases including MIDI, or even games that do not use notes.  You can implement onTile(index) to call noteOn(index).

Reply:
Done!